File: 04-algo/ch-04/1.py
def factorial_r(x):
    if x == 0:
        return 0
    if x == 1:
        return 1

    return x * factorial_r(x - 1)


# don't touch below this line


def test(x):
    res = factorial_r(x)
    print(f"x: {x}")
    print(f"x!: {res}")
    print("====================================")


def main():
    for i in range(1, 15):
        test(i)


main()

File: 09-learn-http/ch-05/1.js
try {
  printCharacterStats(4);
  printCharacterStats("ten");
  printCharacterStats(10);
} catch (error) {
  console.log(error);
}

// don't touch below this line

function printCharacterStats(level) {
  if (isNaN(level)) {
    throw new Error("Parameter is not a number!");
  }
  console.log(`Your character is level ${level}!`);
}
